packages/components/bucket-policy/bucket-policy-body.tsx

```diff
@@ -0,0 +1,442 @@
+          )}
+          onClick={() =>
+            dispatch({
+              type: BucketPolicyActionType.SET_RESOURCE_TYPE,
+              payload: DataResourceType.MULTI,
+            })
+          }
+          checked={!isSingleDataReource}
+        />
+        {!isSingleDataReource &&
+          (showDataResource ? (
+            <MultiDataResource
+              data={dataResources}
+              state={state}
+              launchModal={launchModal}
+              dispatch={dispatch}
+            />
+          ) : (
+            <StatusBox
+              loadError={dataResourcesError}
+              loaded={dataResourcesLoaded}
+            />
+          ))}
+      </FormGroup>
+      <FormGroup fieldId="namespace-name" label={t('Namespace')} isRequired>
+        <p className="pf-c-form__helper-text">
+          {t(
+            'To allow access to this bucket an ObjectBucketClaim will be created in a specific namespace in the OpenShift cluster. Select the namespace in which ObjectBucketClaim will be created'
+          )}
+        </p>
+        <GenericDropdown
+          id="namespace-name"
+          onChangeSelect={(selection) =>
+            dispatch({
+              type: BucketPolicyActionType.SET_OBC_NAMESPACE,
+              payload: selection,
+            })
+          }
+          resource={projectResource}
+          selections={state.obcNamespace}
+        />
+      </FormGroup>
+      <FormGroup fieldId="replication" label={t('Replication')}>
+        <p className="pf-c-form__helper-text">
+          {t(
+            'If you want to replicate data from this bucket to other buckets, select buckets you want to replicate data to'
+          )}
+        </p>
+        <GenericDropdown
+          id="replication"
+          variant={SelectVariant.checkbox}
+          // onChangeSelectOps={onChangeReplication(replicationObj, dispatch)}
+          onChangeSelectOps={onChangeReplication}
+          resource={bucketClaimResource}
+          selections={replicationOBCs}
+          secondaryTextGenerator={getNamespace}
+        />
+      </FormGroup>
+    </Form>
+  );
+};
+
+type onChangeMultiSelectProps = (
+  data: string[],
+  dispatch: React.Dispatch<BucketPolicyAction>
+) => (selection: string) => void;
+
+type onChangeReplicationSelectProps = (resource: K8sResourceKind) => void;
+
+type DataResourceProps = {
+  data: NamespaceStoreKind[];
+  state: BucketPolicyState;
+  launchModal?: LaunchModal;
+  dispatch: React.Dispatch<BucketPolicyAction>;
+};
+
+type DataResourceDropdownProps = Omit<
+  DataResourceProps,
+  'dispatch' | 'state' | 'launchModal'
+> & {
+  launchModal?: () => void;
+  onChange: (selection: string) => void;
+  selections: string | string[];
+  disableAllExcept?: string[];
+  className?: string;
+  variant?: SelectVariant;
+};
+
+type BucketPolicyBodyProps = {
+  ns: string;
+  state: BucketPolicyState;
+  dispatch: React.Dispatch<BucketPolicyAction>;
+  launchModal?: LaunchModal;
+};
```

Comment:
I feel this file can be broken into multiple files, bucket-policy-body.tsx, single-data-resource.tsx, multi-data-resource.tsx

Reply:
ack...